### ISSUE TITLE
PHP has problems after installing the "gd" extension

### ISSUE THREAD

**Issue by CodeKJ** [mlocati/docker-php-extension-installer#537]
### Version of install-php-extensions

1.4.25

### Error description

After versions `1.4.24` and `1.4.25` there are problems running:
```
RUN chmod +x /usr/local/bin/install-php-extensions && sync && \
install-php-extensions gd xdebug
```

Error for version `1.4.24`
```
Check if the zend_extension xdebug can be loaded... Error loading the "xdebug" extension from "/usr/local/lib/php/extensions/no-debug-non-zts-20200930/xdebug.so":
```

Error for version `1.4.25`
```
PHP has problems after installing the "gd" extension:
```

But both `gd` and `xdebug` extensions were working correctly before. From what I see, there are no more details given for this issue.

There are no issues in version `1.4.23`

### Docker image

php:8.0.14-fpm-alpine3.14

### Minimal Dockerfile

```Dockerfile
ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
RUN chmod +x /usr/local/bin/install-php-extensions && sync && \
    install-php-extensions gd xdebug
```

**Comment by svichka** [comment]
I have (I think) the same issue with docker image php:8.1-fpm. While it is installing gd, it tries to re-run make 100 times and exits with an error ```#10 103.4 [0/1] Re-running CMake...
#10 103.5 --- aom_configure: Detected CPU: x86_64
#10 104.0 -- Configuring done
#10 104.1 -- Generating done
#10 104.1 -- Build files have been written to: /tmp/src/tmp.asTJhnjwXE/my.build
#10 104.1 ninja: error: manifest 'build.ninja' still dirty after 100 tries
#10 104.1```

**Comment by CodeKJ** [comment]
In my case looks like it is caused by warning from other extension.

```
#10 19.00 PHP has problems after installing the "gd" extension:
#10 19.00
#10 19.00 Warning: Version warning: Imagick was compiled against ImageMagick version 1803 but version 1808 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
```

But, there was same warning before `1.4.24` and it was fine. Also, `imagick` is not installed by `install-php-extensions`, which probably can be changed to fix this issue, but in this case I am not sure if `install-php-extensions` shouldn't allow this warning, it doesn't hurt overall build. At least in my scenario.

You can reproduce this issue with this dockerfile:
```dockerfile
FROM php:8.0.14-fpm-alpine3.14

RUN apk add --update --upgrade --no-cache --virtual .build-deps $PHPIZE_DEPS \
        imagemagick \
        imagemagick-dev

# this causes warning, but in my case is used to install some other extensions which I have removed from this example.
RUN echo http://dl-2.alpinelinux.org/alpine/edge/community/ >> /etc/apk/repositories

RUN printf "\n" | pecl install \
		imagick && \
		docker-php-ext-enable --ini-name 20-imagick.ini imagick

ADD https://github.com/mlocati/docker-php-extension-installer/releases/download/1.4.25/install-php-extensions /usr/local/bin/
RUN chmod +x /usr/local/bin/install-php-extensions && sync && \ 
		install-php-extensions gd xdebug
```

**Comment by mlocati** [comment]
> I have (I think) the same issue with docker image php:8.1-fpm. While it is installing gd, it tries to re-run make 100 times and exits with an error

@svichka: no, that's unrelated. Could you please open a new issue with the relevant parts of your Dockerfile?

> In my case looks like it is caused by warning from other extension.

@CodeKJ the problem with your approach is that you
- install the imagemagick apk library
- compile the imagick PHP extension (using that imagemagick version)
- you add the edge repository, and imagemagick is upgraded to a newer version

So, your imagick PHP extension is rather unstable.

Whereas you should fix your approach, I agree that `install-php-extensions` shouldn't break your script in this case.

This is fixed by #541 (where the script stops the execution only if installing a PHP extension introduces new warnings.

**Comment by mlocati** [comment]
@CodeKJ your issue is also caused by #416

We have to run `apk upgrade` in order to fix #385.

I think that we could solve your issue (and #416) by avoiding upgrading all the installed apk packages, but only the problematic ones (`wget` at the moment, but probably other ones that may arise in the future).
